File: app/src/main/java/com/simplyillustrate/simplyillustrate/activity/PracticeQuestionDetails.java
package com.simplyillustrate.simplyillustrate.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.simplyillustrate.simplyillustrate.R;

public class PracticeQuestionDetails extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_practice_question_details);
//        Bundle bundle = getIntent().getExtras();
//        PracticeQuestion question = bundle.getParcelable("practiceQuestion");
//        Log.d("QuestionDetails", question.getQuestion());
//        TextView title = findViewById(R.id.question_title);
//        TextView answer1 = findViewById(R.id.anwser_1);
//        TextView answer2 = findViewById(R.id.anwser_2);
//        TextView answer3 = findViewById(R.id.anwser_3);
//        TextView answer4 = findViewById(R.id.anwser_4);
//        ArrayList<String> answers = new ArrayList<>();
//        answers.addAll(question.getIncorrectAnswers());
//        answers.add(question.getCorrectAnswer());
//        Collections.shuffle(answers);
//        title.setText(getString(R.string.question) + question.getQuestion());
//        answer1.setText(getString(R.string.one) + answers.get(0));
//        answer2.setText(getString(R.string.two) + answers.get(1));
//        answer3.setText(getString(R.string.three) + answers.get(2));
//        answer4.setText(getString(R.string.four) + answers.get(3));

        Intent intent = getIntent();
        Bundle bundle = intent.getExtras();
        System.out.println(bundle);

        if (bundle != null)
            System.out.println(bundle.getParcelable("question").toString());
    }
}

File: app/src/main/java/com/simplyillustrate/simplyillustrate/activity/PracticeQuestionDetails.kt
package com.simplyillustrate.simplyillustrate.activity

import android.graphics.Color
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.simplyillustrate.simplyillustrate.R
import com.simplyillustrate.simplyillustrate.entity.PracticeQuestion
import kotlinx.android.synthetic.main.activity_practice_question_details.*
import java.util.*

class PracticeQuestionDetails : AppCompatActivity() {

    var question: PracticeQuestion? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_practice_question_details)

        val bundle = intent.extras
        if (bundle != null) {
            question = bundle.getParcelable<PracticeQuestion>("question")
            val answers = ArrayList<String>()
            answers.addAll(question?.incorrectAnswers!!)
            answers.add(question?.correctAnswer!!)
            answers.shuffle()
            question_title.text = getString(R.string.question) + question?.question!!

            answer1.text = getString(R.string.one) + answers[0]
            answer2.text = getString(R.string.two) + answers[1]
            answer3.text = getString(R.string.three) + answers[2]
            answer4.text = getString(R.string.four) + answers[3]
        }
    }

    fun submitAnswer(view: View) {

        val ans = (view as TextView).text.toString()
        Log.i("PracticeQuestionDetails", ans)
        Log.i("PracticeQuestionDetails", question?.correctAnswer.toString())

        var message = "Incorrect Answer."
        if (question?.correctAnswer == ans.substring(2)) {
            Log.i("PracticeQuestionDetails", "Correct Answer")
            message = "Correct Answer."

            view.setBackgroundColor(Color.GREEN)
        } else
            view.setBackgroundColor(Color.RED)

        Toast.makeText(this@PracticeQuestionDetails, message, Toast.LENGTH_SHORT).show()
    }
}